libpod/top.go

```diff
@@ -0,0 +1,37 @@
+package libpod
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/utils"
+	"github.com/sirupsen/logrus"
+)
+
+// This can be removed is stats is merged
+var CGroupParent = "/libpod_parent"
+
+// GetContainerPids reads sysfs to obtain the pids associated with the container's cgroup
+func (c *Container) GetContainerPids() ([]string, error) {
+	taskFile := filepath.Join("/sys/fs/cgroup/pids", CGroupParent, fmt.Sprintf("libpod-conmon-%s", c.ID()), c.ID(), "tasks")
+	logrus.Debug("reading pids from ", taskFile)
+	content, err := ioutil.ReadFile(taskFile)
+	if err != nil {
+		return []string{}, errors.Wrapf(err, "unable to read pids from %s", taskFile)
+	}
+	return strings.Split(string(content), "\n"), nil
+}
+
+// GetContainerPidInformation calls ps with the appropriate options and returns
+// the results as a string
+func (c *Container) GetContainerPidInformation(args []string) (string, error) {
```

Comment:
Can we return an array, with one line for each PID we're monitoring?

Reply:
Ideally, we'd parse the output a tiny bit, so we could return something easily machine-parsable

Reply:
Further thought: maybe we should also hardcode the list of PIDs we gets stats from if we modify the return. We can get stats on all PIDs in the container, and kpod can throw away the ones it doesn't need. It just seems a little dodgy to me to let the user give us an arbitrary list of PIDs

Reply:
The user cannot provide a list of pids .. The pids are determined by GetContainePids.

Reply:
And I see no reason to parse the output ... we are following what docker did and they dump it.  There are no JSON options.

Reply:
Well, less "user" and more "API consumer" - I'd like to move getting the PIDs into here so you don't need to manually add them to the ps args.

Reply:
I think i would prefer outputting an array of strings, with each string being a single line of the output of PS. You can then do a contains on those lines in kpod ps to grab on the PIDs you want. The first line might need special handling as it has the column names.